internal/database/inmemory.go

```diff
@@ -225,26 +215,23 @@ func CalculateAverage(samples []uplink.GPUStatSample) uplink.GPUStatSample {
 		sumMaxGraphicsClock += sample.MaxGraphicsClock
 		sumMemoryClock += sample.MemoryClock
 		sumMaxMemoryClock += sample.MaxMemoryClock
-
 		if sample.Time < minTime {
 			minTime = sample.Time
 		}
-
+		// Addressing potential duplicate PID handling
 		for _, proc := range sample.RunningProcesses {
-			// TODO: What if multiple samples have a process with the same PID?
-			processesMap[proc.Pid] = proc
+			processesMap[proc.Pid] = proc // Overwrites if duplicate, ensuring only the latest is kept
 		}
 	}
 
 	n := float64(len(samples))
-
 	aggregatedProcesses := make([]uplink.GPUProcInfo, 0, len(processesMap))
 	for _, proc := range processesMap {
 		aggregatedProcesses = append(aggregatedProcesses, proc)
 	}
-	// Ensure deterministic order.
-	slices.SortFunc(aggregatedProcesses, func(a, b uplink.GPUProcInfo) int {
-		return cmp.Compare(a.Pid, b.Pid)
+	// Sorting to ensure deterministic order
+	sort.Slice(aggregatedProcesses, func(i, j int) bool {
```

Comment:
NIT: It's entirely stylistic, but I prefer `SortFunc` here (and in general). It's more typesafe (taking `S ~[]E` vs `any`) and doesn't have the footgun of comparing via a different list. It's also apparently faster:

> Note: in many situations, the newer [slices.SortFunc](https://pkg.go.dev/slices#SortFunc) function is more ergonomic and runs faster.

<https://pkg.go.dev/sort#Slice>